templates/prometheus.go

```diff
@@ -45,5 +53,26 @@ var PrometheusTemplate = promv1.Prometheus{
 			}},
 		},
 		Resources: utils.GetResourceRequirements("prometheus"),
+		Web:       PrometheusWebSpecTemplate.DeepCopy(),
+	},
+}
+
+var locSecretRef = corev1.LocalObjectReference{
+	Name: PrometheusServiceSecretName,
+}
+
+var PrometheusWebSpecTemplate = &promv1.WebSpec{
+	TLSConfig: &promv1.WebTLSConfig{
+		KeySecret: corev1.SecretKeySelector{
+			LocalObjectReference: locSecretRef,
+			Key:                  "tls.key",
+		},
+		Cert: promv1.SecretOrConfigMap{
+			Secret: &corev1.SecretKeySelector{
+				LocalObjectReference: locSecretRef,
+				Key:                  "tls.crt",
+			},
+		},
+		ClientAuthType: fmt.Sprint(tls.VerifyClientCertIfGiven),
 	},
```

Comment:
This should also go into the template:
```suggestion
		ClientCA: promv1.SecretOrConfigMap{
			ConfigMap: &corev1.ConfigMapKeySelector{
				LocalObjectReference: corev1.LocalObjectReference{Name: ""},
				Key: "service-ca.crt",
			},
		},
	},
```

Reply:
done